src/main/java/fr/openent/competences/service/impl/DefaultBFCService.java

```diff
@@ -542,103 +547,114 @@ public void buildBFC(final boolean recapEval, final String[] idEleves, final Str
+                    echelleConversionFuture.failed();
+                    handler.handle(new Either.Left<>("Impossible de recuperer l'echelle de conversion pour la " +
+                            "classe selectionnee :\n" + event.left().getValue()));
+                    log.error("buildBFC : getEchelleConversion : " + event.left().getValue());
                 }
-                getMaxNoteCompetenceEleveFuture.complete();
-            } else {
-                getMaxNoteCompetenceEleveFuture.failed();
-                handler.handle(new Either.Left<>("Impossible de recuperer les evaluations pour " +
-                        "la classe selectionnee :\n" + event.left().getValue()));
-                log.error("buildBFC : getMaxNoteCompetenceEleve : " + event.left().getValue());
-            }
-        });
-
-        Future echelleConversionFuture = Future.future();
-        listOfFutures.add(echelleConversionFuture);
-        getEchelleConversion(idStructure, idClasse, event -> {
-            if (event.isRight()) {
-                echelleConversion.addAll(event.right().getValue());
-                echelleConversionFuture.complete();
-            } else {
-                echelleConversionFuture.failed();
-                handler.handle(new Either.Left<>("Impossible de recuperer l'echelle de conversion pour la " +
-                        "classe selectionnee :\n" + event.left().getValue()));
-                log.error("buildBFC : getEchelleConversion : " + event.left().getValue());
-            }
-        });
+            });
 
-        Future domainesFuture = Future.future();
-        listOfFutures.add(domainesFuture);
-        getDomaines(idClasse, idCycle, event -> {
-            if (event.isRight()) {
-                Set<Domaine> setDomainesRacine = new LinkedHashSet<>();
+            Future domainesFuture = Future.future();
+            listOfFutures.add(domainesFuture);
+            getDomaines(idClasse, idCycle, event -> {
+                if (event.isRight()) {
+                    Set<Domaine> setDomainesRacine = new LinkedHashSet<>();
 
-                final Map<Long, Domaine> domaines = new HashMap<>();
-                domaines.putAll(event.right().getValue());
-                for (Domaine domaine : domaines.values()) {
-                    if (domaine.getParentRacine() != null) {
-                        setDomainesRacine.add(domaine.getParentRacine());
+                    final Map<Long, Domaine> domaines = new HashMap<>();
+                    domaines.putAll(event.right().getValue());
+                    for (Domaine domaine : domaines.values()) {
+                        if (domaine.getParentRacine() != null) {
+                            setDomainesRacine.add(domaine.getParentRacine());
+                        }
                     }
+                    domainesRacine.addAll(setDomainesRacine);
+
+                    domainesFuture.complete();
+                } else {
+                    domainesFuture.failed();
```

Comment:
`  /**
   * Did it fail?
   *
   * @return true if it failed or false otherwise
   */
  boolean failed();`

Du coups cette ligne ne sert a rien